xbmc/guilib/GUIFontManager.cpp

```diff
@@ -142,37 +153,41 @@ CGUIFont* GUIFontManager::LoadTTF(const std::string& strFontName,
   }
 
   // check if we already have this font file loaded (font object could differ only by color or style)
-  std::string TTFfontName =
+  const std::string fontIdent =
       StringUtils::Format("{}_{:f}_{:f}{}", strFilename, newSize, aspect, border ? "_border" : "");
 
-  CGUIFontTTF* pFontFile = GetFontFile(TTFfontName);
+  std::shared_ptr<CGUIFontTTF> pFontFile = GetFontFile(fontIdent);
   if (!pFontFile)
   {
-    pFontFile = CGUIFontTTF::CreateGUIFontTTF(TTFfontName);
+    pFontFile = CGUIFontTTF::CreateGUIFontTTF(fontIdent);
     bool bFontLoaded = pFontFile->Load(strPath, newSize, aspect, 1.0f, border);
 
     if (!bFontLoaded)
     {
-      delete pFontFile;
-
       // font could not be loaded - try Arial.ttf, which we distribute
       if (strFilename != "arial.ttf")
       {
-        CLog::Log(LOGERROR, "Couldn't load font name: {}({}), trying to substitute arial.ttf",
-                  strFontName, strFilename);
-        return LoadTTF(strFontName, "arial.ttf", textColor, shadowColor, iSize, iStyle, border, lineSpacing, originalAspect);
+        CLog::Log(
+            LOGERROR,
+            "GUIFontManager::{}: Couldn't load font name: {}({}), trying to substitute arial.ttf",
+            __func__, strFontName, strFilename);
+        return LoadTTF(strFontName, "arial.ttf", textColor, shadowColor, iSize, iStyle, border,
+                       lineSpacing, originalAspect);
       }
-      CLog::Log(LOGERROR, "Couldn't load font name:{} file:{}", strFontName, strPath);
+      CLog::Log(LOGERROR, "GUIFontManager::{}: Couldn't load font name:{} file:{}", __func__,
+                strFontName, strPath);
 
-      return NULL;
+      return nullptr;
     }
 
-    m_vecFontFiles.push_back(pFontFile);
+    m_vecFontFiles.emplace_back(pFontFile);
   }
 
   // font file is loaded, create our CGUIFont
-  CGUIFont *pNewFont = new CGUIFont(strFontName, iStyle, textColor, shadowColor, lineSpacing, (float)iSize, pFontFile);
-  m_vecFonts.push_back(pNewFont);
+  std::shared_ptr<CGUIFont> pNewFont =
```

Comment:
good candidate for `auto`

Reply:
Makes it shorter, good idea and clang is happy.